src/odc/nc2odb.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python
+
+import sys
+import os
+from pathlib import Path
+prefix_lib_path = Path(__file__).absolute().parent.parent/'lib'
+prefix_bin_path = prefix_lib_path.parent/'bin'
+sys.path.append(str(prefix_lib_path))
+sys.path.append(str(prefix_lib_path/'python{}.{}'.format(*sys.version_info)/'site-packages'))
+os.environ['PATH'] = "%s:%s" % (os.environ['PATH'], str(prefix_bin_path))
+
+import tempfile
+import argparse
+import datetime as dt
+import netCDF4 as nc
+import codyssey as odc
+
+###################################################################################
+# SUBROUTINES
+###################################################################################
+
+
+def CharArrayToStrings(CharArray):
+    Strings = []
+    for ByteList in CharArray:
+        Strings.append(b''.join(ByteList).decode('ascii'))
+    return Strings
+
+
+def ConvertDtStrings(DtStrings):
+    Dates = []
+    Times = []
+    for DtStr in DtStrings:
+        Dt = dt.datetime.strptime(DtStr, "%Y-%m-%dT%H:%M:%SZ")
+        (Date, Time) = DtToDateTime(Dt)
+
+        Dates.append(Date)
+        Times.append(Time)
+
+    return (Dates, Times)
+
+
+def ConvertDtRefOffset(RefDtInt, OffsetTime):
+    # convert the reference date time to a datetime object
+    TempRef = RefDtInt
+    Year = TempRef // 1000000
+    TempRef = TempRef % 1000000
+    Month = TempRef // 10000
+    TempRef = TempRef % 10000
+    Day = TempRef // 100
+    Hour = TempRef % 100
+    RefDt = dt.datetime(Year, Month, Day, Hour)
+
+    Dates = []
+    Times = []
+    for Otime in OffsetTime:
+        OffsetSeconds = round(3600 * Otime)
+        OffsetDt = dt.timedelta(seconds=OffsetSeconds)
+        Dt = RefDt + OffsetDt
+        (Date, Time) = DtToDateTime(Dt)
+
+        Dates.append(Date)
+        Times.append(Time)
+
+    return (Dates, Times)
+
+
+def DtToDateTime(Dt):
+    Date = Dt.year * 10000
+    Date += Dt.month * 100
+    Date += Dt.day
+
+    Time = Dt.hour * 10000
+    Time += Dt.minute * 100
+    Time += Dt.second
+
+    return (Date, Time)
+
+
+###################################################################################
+# MAIN
+###################################################################################
+ScriptName = os.path.basename(sys.argv[0])
+
+# Parse command line
+ap = argparse.ArgumentParser()
+ap.add_argument("input_netcdf", help="path to input netCDF4 file")
```

Comment:
If we passed a list of netcdf files, we could thread/parallelize it.  This could speed up when processing a directory of files.
May be another time, or if you want to wait, I can provide a parallelized/threaded patch on Monday.

Reply:
The run_nc2odb.sh companion script creates a list of netcdf files and runs nc2odb.py on individual files. This way, the list of netcdf files can be gigantic and not overflow the capacity of wildcard expansion in the shell. Would it make more sense to parallelize the run_nc2odb.sh script instead of the nc2odb.py script?

Reply:
Either way is fine.  Parallelize internally, such that each thread/process handles one file, or externally through the shell script.  The latter implies, all threads/processes need to be completed till the next batch of threads/processes start. 

To avoid overflow, you have to cap the number of parallel processing; be it shell or in the python.

We can leave this as is and re-visit if we find that we are waiting for a while on this converter.  How does that sound?

Reply:
I view this as something a higher-level workflow should be doing.  The `obs-convert` app in JEDI-Rapids is currently using process-based parallelism with to run this script when converting.

Reply:
I think we are good for now, and we can address this in the future if the need arises.

I'm going to go ahead and merge now. Thanks for the reviews!